docs/proposals/README.md

```diff
@@ -0,0 +1,105 @@
+- Document design ideas, tradeoffs, and decisions for historical reference.
+
+[gep]: https://github.com/kubernetes-sigs/gateway-api/blob/c8b54a05c850cd717eb852c4874c6c89d02a5ef8/geps/overview.md
+
+## When to Write an Enhancement Proposal
+
+You should only write an Enhancement Proposal if a maintainer has requested one for a particular issue. All enhancement
+requests should start as an idea on [GitHub Discussions][discussion]. Not all enhancement requests will require an
+Enhancement Proposal. For example, here are some examples of requests that may not need an Enhancement Proposal:
+
+* Gateway API fields. However, some larger Gateway API fields may require Enhancement Proposals if they require
+  significant changes to the architecture of the code.
+* Small changes (validation, documentation, fixups). It is always possible that the maintainers will determine a "small"
+  change ends up requiring a Enhancement Proposal.
+
+[discussion]: https://github.com/nginxinc/nginx-kubernetes-gateway/discussions
+
+## Process
+
+The diagram below shows the Enhancement Proposal process:
+
+```mermaid
+flowchart TD
+    D([Open Discussion]) --> C
+    C([Issue Created]) --> Provisional
+    Provisional -->|Enhancement Proposal<br /> Doc PR done| Implementable
+    Implementable -->|Work completed| Completed
+```
+
+### 1. Open a GitHub Discussion
+
+Before creating an issue or Enhancement Proposal, [open an idea][idea] on GitHub discussion. Describe the enhancement
+you would like, any use cases you have, and other relevant details. Beginning with a discussion allows you to get
+feedback from the maintainers and the community before you invest time in writing an Enhancement Proposal.
+
+[idea]: https://github.com/nginxinc/nginx-kubernetes-gateway/discussions/new?category=ideas
+
+### 2. Create an Issue
+
+If there is consensus on the discussion post that the enhancement is important and should be included in the roadmap, a
+maintainer will ask you to [open an issue][issue] on GitHub.
+
+Not every enhancement warrants an Enhancement Proposal. _If_ the enhancement issue requires an Enhancement Proposals,
+the maintainers will add the label `needs-enhancement-proposal` to the issue.
+
+[issue]: https://github.com/nginxinc/nginx-kubernetes-gateway/issues/new?assignees=&labels=proposal&projects=&template=enhancement.md&title=
+
+### 3. Agree on the Goals (Provisional)
+
+Write the first version of your Enhancement Proposal using the [template](/docs/proposals/template.md), including only
+the summary of the enhancement and the sections addressing the "Goals" and "Non-Goals". The purpose of this initial
+Enhancement Proposal is to achieve consensus on the objectives before filling out the details of implementation. Set the
+[status](#status) field in the Enhancement Proposal document to "Provisional".
+
+Open a Pull Request with your Enhancement Proposal and work with the reviewers to get the necessary approvals.
```

Comment:
Does this mean that all enhancement proposals should be pull requests with a corresponding issue (linked in the template)? Meaning, the user is required to change something in the repository in order to write an enhancement proposal? Is this done in a certain directory (i.e. an enhancement proposal directory)? Should the user create a new document in the repository that contains their enhancement proposal with their pull request?

If so, we may want to rewrite this to be more explicit as to what actions are required by the submitter in this case.

Reply:
Yes to all of your questions. 

Is that not explicit in this doc? In the steps where a PR is needed that is explicitly called out, as is the directory where they are expected to go. 

I think the confusion is that the process outlined in this document will only be followed when a maintainer tags an issue as `needs-enhancment-proposal`. 

There will be plenty of times that an issue is sufficient, and we won't require a user to go through this process. This document is describing the process of writing and implementing an enhancement proposal. An enhancement proposal is a document that is written and _committed_ to the repository.

Reply:
I think adding a section at the top of this doc describing exactly what an enhancement proposal is will help.